### ISSUE TITLE
"files_sharing.sharecontroller.showShare\" must match \"[^\/]++\" (\"\" given) to generate a corresponding URL

### ISSUE THREAD

**Issue by enoch85** [owncloud/core#14231]
### Steps to reproduce

Don´t know.

### Expected behavior

No errors

### Actual behavior

(You should make new lines in the downloadable owncloud.log)

```
{"reqId":"024fb12f597062233005cfe8a2fac3ed","remoteAddr":"87.230.85.166","app":"remote","message":"Parameter \"token\" for route \"files_sharing.sharecontroller.showShare\" must match \"[^\/]++\" (\"\" given) to generate a corresponding URL.","level":4,"time":"2015-02-15T07:57:20+00:00"},
{"reqId":"9fcd82882ff0a66d0363d767672e4b5d","remoteAddr":"87.230.85.166","app":"remote","message":"Parameter \"token\" for route \"files_sharing.sharecontroller.showShare\" must match \"[^\/]++\" (\"\" given) to generate a corresponding URL.","level":4,"time":"2015-02-15T07:57:20+00:00"},
{"reqId":"d25d8dcbcc2a9b3af5015c5f91d9d607","remoteAddr":"87.230.85.166","app":"remote","message":"Parameter \"token\" for route \"files_sharing.sharecontroller.showShare\" must match \"[^\/]++\" (\"\" given) to generate a corresponding URL.","level":4,"time":"2015-02-15T07:57:19+00:00"},
{"reqId":"75b464ad096e6b016582ba3e45a2dcc9","remoteAddr":"5.255.253.4","app":"remote","message":"Parameter \"token\" for route \"files_sharing.sharecontroller.showShare\" must match \"[^\/]++\" (\"\" given) to generate a corresponding URL.","level":4,"time":"2015-02-15T07:57:18+00:00"},
{"reqId":"f8cef19450d39f486916667e119e9fac","remoteAddr":"87.230.85.166","app":"remote","message":"Parameter \"token\" for route \"files_sharing.sharecontroller.showShare\" must match \"[^\/]++\" (\"\" given) to generate a corresponding URL.","level":4,"time":"2015-02-15T07:57:17+00:00"},
```

### Server configuration

Ubuntu Server 12.04

**Database:**
Server version: 5.5.41-0ubuntu0.12.04.1 - (Ubuntu)
PHP version:
PHP 5.5.9-1ubuntu4.5 (cli) (built: Oct 29 2014 11:59:10)

**ownCloud version: (see ownCloud admin page)**
8.0
*Updated from an older ownCloud or fresh install:*
5, 6, 7.

**List of activated apps:**
Activity
Calender 
Contacts
Deleted files
Documents
First Run Wizard
Full Text Search
PDF Viewer
Pictures
Share Files
Text Editor
Updater
Versions
Video Viewer

**Are you using external storage, if yes which one: local/smb/sftp/...**
No
**Are you using encryption: yes/no**
No

### Client configuration

**Browser:**
Version 39.0.2171.71 m
**Operating system:**
Windows 8.1
**Sync Client**
1.7.1 stable

### Logs

**Web server error log**

```
Insert your webserver log here
```

**ownCloud log (data/owncloud.log)**

```
As mentioned above
```

**Browser log**

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log 
c) ...
```

**Comment by karlitschek** [comment]
@schiesbn

**Comment by DeepDiver1975** [comment]
> (You should make new lines in the downloadable owncloud.log)

isolated issue please - Thanks a lot

**Comment by DeepDiver1975** [comment]
> Steps to reproduce
> 
> Don´t know.

hmmm - hard to tell how this couod have happened

**Comment by enoch85** [comment]
@DeepDiver1975 
> isolated issue please - Thanks a lot

Done #14268 

> hmmm - hard to tell how this couod have happened

I figured that maybe the log would give you some info.

**Comment by PVince81** [comment]
Sounds like a public share that has no token ?
Did you or someone try opening a public link but stripped out the token ?

**Comment by PVince81** [comment]
Reproductible by doing this on OC8:
- open "http://localhost/owncloud/public.php?service=files" (without token)
or
- open "http://localhost/owncloud/public.php?service=files&t=1e357dff477ae09bb97e6023160b4999/" (with a trailing slash)

I suspect it might also happen if the token contains invalid chars.
Could be due to a failed copy paste from a user ?

**Comment by enoch85** [comment]
@PVince81 Actually, I removed all my shares when installing oC8 by deleteing the cache and share tables in MySQL. After that I re-shared everything. So that's not likely the issue here.

Or do you mean that people are trying to open old links?

**Comment by enoch85** [comment]
Is this related?

```
[Tue Feb 17 21:39:10 2015] [error] [client 192.168.1.1] PHP Notice:  iconv(): Detected an illegal character in input string in /var/www/owncloud/apps/search_lucene/3rdparty/zendframework/zendsearch/library/ZendSearch/Lucene/Document/HTML.php on line 111, referer: https://en0ch.se/index.php/s/9ytZ67M89vLRL45
[Tue Feb 17 21:39:10 2015] [error] [client 192.168.1.1] PHP Notice:  iconv(): Detected an illegal character in input string in /var/www/owncloud/apps/search_lucene/3rdparty/zendframework/zendsearch/library/ZendSearch/Lucene/Document/HTML.php on line 111, referer: https://en0ch.se/index.php/s/9ytZ67M89vLRL45
```

**Comment by PVince81** [comment]
Hmm, in your case the error said `(\"\" given)` which looks like the route parser is getting an empty string ?!

I tried removing the token from the URL, but that would cause a redirect to the files app.
So I have no clue how the router managed to get itself in this state.

I see there are a few redirect stuff in the code, maybe somehow the token got mangled at this point (or are these null bytes??)

@LukasReschke any clue ?

**Comment by LukasReschke** [comment]
That happens when somebody accesses http://localhost/master/public.php?service=files&t= where the token is then undefined.

**Comment by LukasReschke** [comment]
See https://github.com/owncloud/core/blob/stable8/apps/files_sharing/public.php#L17

**Comment by LukasReschke** [comment]
Preparing patch…

**Comment by LukasReschke** [comment]
Patch is at https://github.com/owncloud/core/pull/14509 - please test.

**Comment by PVince81** [comment]
@enoch85 it is likely that some users have opened links that had no token in them. This could happen when email clients break the URLs badly. I don't think this is critical, defer to 8.1 ? @DeepDiver1975

**Comment by PVince81** [comment]
With "defer" I mean "no backport". There is already a fix :smile:

**Comment by LukasReschke** [comment]
Agreed with @PVince81. Technically not highly critical bug so IMHO no backport? – Unless our policy is "Critical bugs as well as no-burners"?

**Comment by enoch85** [comment]
@LukasReschke Thanks for the fix! And Thanks for testing @PVince81

**Comment by LukasReschke** [comment]
Thank you for reporting it :smile:

**Comment by enoch85** [comment]
Thought this should be released with 8.0.1?

**Comment by PVince81** [comment]
It's not a critical bug and more a corner case, so more like 8.1.

**Comment by mbach04** [comment]
Hello everyone,

I've run into this same issue with an 8.0 install of Owncloud via a EC2 instance on Amazon Web Services. The install comes ready to rock from Bitnami, it's running the required PHP as well. However, creating a share link via curl with the following command:
```
curl "https://mydomain.compute-1.amazonaws.com/ocs/v1.php/apps/files_sharing/api/v1/shares" -k -u $USER:$PASS -X POST --data "path=ownCloudUserManual.pdf&shareType=3"
```
Returns the following:
```
<?xml version="1.0"?>
<ocs>
 <meta>
  <status>ok</status>
  <statuscode>100</statuscode>
  <message/>
 </meta>
 <data>
  <id>9</id>
  <url>https://mydomain.compute-1.amazonaws.com/public.php?service=files&amp;t=Nka0PHp2AKlK73T</url>
  <token>Nka0PHp2AKlK73T</token>
 </data>
</ocs>
```
When I try to access the link provided I receive the "INTERNAL SERVER ERROR" Splash page.

Inspection of the Owncloud log shows the following.
```
Parameter "token" for route "files_sharing.sharecontroller.showShare" must match "[^/]++" ("" given) to generate a corresponding URL.
```

Is this fixed by upgrading to 8.0.2? 

Any thoughts?

Thanks for the help!

**Comment by enoch85** [comment]
@mbach04 This will be fixed in 8.1.
